### PR TITLE
aoc: Migrate to use cmif serialization

### PR DESCRIPTION
**Pull request by FearlessTobi**

### CODE REVIEW

File: src/core/hle/service/aoc/aoc_u.cpp [hunk 1]
```diff
@@ -46,104 +46,28 @@ static std::vector<u64> AccumulateAOCTitleIDs(Core::System& system) {
...
-        IPC::ResponseBuilder rb{ctx, 2};
-        rb.Push(ResultNoPurchasedProductInfoAvailable);
-    }
-
-    KernelHelpers::ServiceContext service_context;
-
-    Kernel::KEvent* purchased_event;
-};
-
 AOC_U::AOC_U(Core::System& system_)
```
Comment (liamwhite) [inline src/core/hle/service/aoc/aoc_u.cpp:49]: This class is IAddOnContentManager (the file should be renamed as well)

File: src/core/hle/service/aoc/aoc_u.cpp
Comment (liamwhite) [inline src/core/hle/service/aoc/aoc_u.cpp:1]: The previous contents of IAddOnContentManager were accidentally left in this file. Also this file which contains the server manager is named aoc_u but should probably just be called aoc now.

**Comment by liamwhite** [comment]
breaks dlc enumeration

File: src/core/hle/service/aoc/addon_content_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        }
+    }
+
+    // TODO(DarkLordZach): Find the correct error code.
+    R_UNLESS(out.size() >= offset, ResultUnknown);
+
+    *out_count = static_cast<u32>(std::min<size_t>(out.size() - offset, count));
+    std::rotate(out.begin(), out.begin() + offset, out.end());
+
+    std::memcpy(out_addons.data(), out.data(), *out_count);
```
Comment (german77) [inline src/core/hle/service/aoc/addon_content_manager.cpp:132]: *out_count * sizeof(u32)

**Comment by FearlessTobi** [comment]
Should be fixed now. Thanks for spotting the bug.

**Review by liamwhite — APPROVED** [review]

**Pull request merged**